packages/eas-cli/src/commands/fingerprint/generate.ts

```diff
@@ -64,12 +74,21 @@ export default class FingerprintGenerate extends EasCommand {
       }
       platform = await selectRequestedPlatformAsync();
     }
+
+    let env: Env | undefined;
+    if (environment) {
+      env = environment
+        ? { ...(await getServerSideEnvironmentVariablesAsync()), EXPO_NO_DOTENV: '1' }
+        : {};
+    }
```

Comment:
```suggestion
    const env = environment
      ? { ...(await getServerSideEnvironmentVariablesAsync()), EXPO_NO_DOTENV: '1' }
      : undefined;
```

Reply:
Also regarding `EXPO_NO_DOTENV`, I see this is the behavior for `eas update` for calling into the expo cli (which I assume fingerprint does under the hood hence why we add it to this call), but is it also the behavior for `eas build`?

Reply:
i passed `EXPO_NO_DOTENV` because it is also passed to grab the config elsewhere in `eas-cli` (npx expo config is a shim over expo/config): https://github.com/expo/eas-cli/blob/22bc21ec9c5571cbbe428e89213e5e459f7de767/packages/eas-cli/src/project/expoConfig.ts#L59-L72

under the hood, the fingerprint library calls into the `expo/config` library  https://github.com/expo/eas-cli/blob/22bc21ec9c5571cbbe428e89213e5e459f7de767/packages/eas-cli/src/project/expoConfig.ts#L59-L72